internal/sourcemap/sourcemapper.go

```diff
@@ -0,0 +1,200 @@
+package sourcemap
+
+import (
+	"strings"
+
+	"github.com/microsoft/typescript-go/internal/core"
+	"github.com/microsoft/typescript-go/internal/tspath"
+)
+
+// SourceMapper provides source mapping functionality for LSP operations
+type SourceMapper interface {
```

Comment:
I don't think all of these interfaces are needed; surely we only have one implementation of most of these and could just use a concrete type? do we need multiple hosts, interfaces, then struct fields of the various kinds?